frontend/www/js/omegaup/components/notification/Clarifications.vue

```diff
@@ -0,0 +1,211 @@
+  background-color: #7f7f7f;
+}
+
+.notification-drawer {
+  width: 320px;
+  max-width: 320px;
+  max-height: 380px;
+  overflow-y: scroll;
+}
+
+.notification-drawer li {
+  padding: 3px 20px;
+  border-top: 1px solid #f1f1f1;
+}
+
+.notification-drawer li a {
+  color: #333;
+  text-decoration: none;
+}
+
+.notification-drawer li a pre {
+  overflow: hidden;
+  text-overflow: ellipsis;
+  white-space: nowrap;
+  width: 100%;
+}
+
+.notification-drawer li:hover,
+.notification-drawer li:focus,
+.notification-drawer li:active {
+  cursor: pointer;
+  background-color: #678dd7;
+  text-decoration: none;
+}
+
+.notification-drawer li:hover > a,
+.notification-drawer li:focus > a,
+.notification-drawer li:active > a {
+  color: #fff;
+}
+
+.notification-drawer li a > h4,
+.notification-drawer li a > p {
+  word-wrap: break-word;
+}
+</style>
+
+<script lang="ts">
+import { Vue, Component, Prop, Watch } from 'vue-property-decorator';
+import { T } from '../../omegaup.js';
+import omegaup from '../../api.js';
+
+@Component
+export default class Clarifications extends Vue {
+  @Prop() initialClarifications!: omegaup.Clarification[];
+  T = T;
+
+  unread: boolean = true;
+  flashInterval: number = 0;
+  clarifications: omegaup.Clarification[] = this.initialClarifications;
+
+  @Watch('initialClarifications')
+  onPropertyChanged(
+    newValue: Array<omegaup.Clarification>,
+    oldValue: Array<omegaup.Clarification>,
+  ): void {
+    this.clarifications = newValue;
+    this.unread = true;
+    const audio = <HTMLMediaElement>(
+      document.getElementById('notification-audio')
+    );
+    if (audio !== null) {
+      audio.play();
+    }
+  }
+
+  @Watch('unread')
+  onPropertyChange(newValue: boolean, oldValue: boolean): void {
+    if (newValue) {
+      if (this.flashInterval) return;
+      this.flashInterval = setInterval(this.flashTitle, 1000);
+    } else {
+      if (!this.flashInterval) return;
+      clearInterval(this.flashInterval);
+      this.flashInterval = 0;
+      if (document.title.indexOf('!') === 0) {
+        document.title = document.title.substring(2);
+      }
+    }
+  }
+
+  anchor(clarification: omegaup.Clarification): string {
+    return `#clarifications/clarification-${clarification.clarification_id}`;
+  }
+
+  flashTitle(reset: boolean): void {
+    if (document.title.indexOf('!') === 0) {
+      document.title = document.title.substring(2);
+    } else if (!reset) {
+      document.title = '! ' + document.title;
+    }
+  }
+
+  onCloseClicked(clarification: omegaup.Clarification): void {
+    const id = `clarification-${clarification.clarification_id}`;
+    this.clarifications = this.clarifications.filter(
+      element => element.clarification_id !== clarification.clarification_id,
+    );
+    localStorage.setItem(id, Date.now().toString());
+  }
+
+  onMarkAllAsRead(): void {
+    for (const key in this.clarifications) {
+      const id = `clarification-${this.clarifications[key].clarification_id}`;
```

Comment:
nit: siempre que puedas, usa [`for...of`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...of):

```suggestion
    for (const clarification of this.clarifications) {
      const id = `clarification-${clarification.clarification_id}`;
```

Reply:
Listo